extensions/transform/org.eclipse.smarthome.transform.jsonpath/src/main/java/org/eclipse/smarthome/transform/jsonpath/internal/JSonPathTransformationService.java

```diff
@@ -65,7 +65,7 @@ public String transform(String jsonPathExpression, String source) throws Transfo
                 return transformationResult.toString();
             }
         } catch (PathNotFoundException e) {
-            throw new TransformationException("Invalid path '" + jsonPathExpression + "' in '" + source + "'");
+            return null;
```

Comment:
I'm not sure, if #4870 addresses the same problem.
Let me describe, why I think, an exception in case of a PathNotFound isn't the right way to handle this, as it is always possible, that a path doesn't exist in a json-Document. The main question seems to ... How to deal with that correctly?

In my case, I'll query the Weather warnings from the DWD. The DWD publishes a json-Document with all known warnings. The json-Document only contains warnings. If no warnings exists for your town or district, it doesn't show up in the json-Document.
If you do now something like the following in a rule, it works, when a warning exsits, but do some strange things, if your town and district is not included in the json-Document. The "old" way was easy to handle, because the variable was NULL in case of a pathNotFound-Exception. Now it includes the whole json-String :-/

For example, this works before the original change

```
var String json = sendHttpGetRequest("https://www.dwd.de/DWD/warnungen/warnapp/json/warnings.json")
var String headline = transform("JSONPATH", "$.warnings.108425000[0].headline", json_1)

if (headline === null) {
     headline = "Keine Unwetterwarnung"
 }
 postUpdate(weather_warning_dwd_headline, headline)
```

Now, I receive an exception in my openhab.log, every the document doesn't contain my path (108425000):

```
2018-02-06 12:39:59.938 [ERROR] [ore.transform.actions.Transformation] - Error executing the transformation 'JSONPATH': Invalid path '$.warnings.108425000[0].headline' in '{"time":1517916800000,"warnings":{"106535000":[{"regionName":"Vogelsbergkreis","start":1517907600000,"end":1517936400000,"level":2,"type":5,"state":"Hessen","description":"Es tritt oberhalb 400 m leichter Frost zwischen 0 °C und -4 °C auf.","headline":"Amtliche WARNUNG vor FROST","event":"FROST","instruction":"","stateShort":"HE","altitudeStart":400,"altitudeEnd":null},{"regionName":"Vogelsbergkreis","start":1517907600000,"end":1517936400000,"level":2,"type":5,"state":"Hessen","description":"Es tritt leichter Frost zwischen 0 °C und -4 °C auf.","headline":"Amtliche WARNUNG vor FROST","event":"FROST","instruction":"","stateShort":"HE","altitudeStart":null,"altitudeEnd":null}],"103401000":[{"regionName":"Stadt Delmenhorst","start":1517914800000,"end":1517936400000,"level":2,"type":5,"state":"Niedersachsen","description":"Es tritt leichter ...
```
So it works like designed in https://github.com/eclipse/smarthome/commit/f90852c9a4d8717b2e86e0deb581a1a34fe8f170

But the main problem in my case is now, the variable "headline" is now filled up with the complete json-document:

```
logInfo("weather.rules","[DWD-Wetter] Meldung: " + headline)
```
```
2018-02-06 12:39:59.944 [INFO ] [smarthome.model.script.weather.rules] - [DWD-Wetter] Meldung: {"time":1517916800000,"warnings":{"106535000":[{"regionName":"Vogelsbergkreis","start":1517907600000,"end":1517936400000,"level":2,"type":5,"state":"Hessen","description":"Es tritt oberhalb 400 m leichter Frost zwischen 0 °C und -4 °C auf.","headline":"Amtliche WARNUNG vor FROST","event":"FROST","instruction":"","stateShort":"HE","altitudeStart":400,"altitudeEnd":null},{"regionName":"Vogelsbergkreis","start":1517907600000,"end":1517936400000,"level":2,"type":5,"state":"Hessen","description":"Es tritt leichter Frost zwischen 0 °C und -4 °C auf.","headline":"Amtliche WARNUNG vor FROST","event":"FROST","instruction":"","stateShort":"HE","altitudeStart":null,"altitudeEnd":null}],"103401000":[{"regionName":"Stadt Delmenhorst","start":1517914800000,"end":1517936400000,"level":2,"type":5,"state":"Niedersachsen","description":"Es tritt leichter Frost zwischen 0 °C und -4 °C auf.","headline":"Amtliche WARNUNG vor FROST",
...
```
And due to this behaviour (not null, in case if it's not existing), the item also gets updated.

```
2018-02-06 12:40:00.177 [vent.ItemStateChangedEvent] - weather_warning_dwd_headline changed from {"time":1517916800000,"warnings":{"106535000":[{"regionName":"Vogelsbergkreis","start":1517907600000,"end":1517936400000,"level":2,"type":5,"state":"Hessen","description":"Es tritt oberhalb 400 m leichter Frost zwischen 0 °C und -4 °C auf.","headline":"Amtliche WARNUNG vor FROST","event":"FROST","instruction":"","stateShort":"HE","altitudeStart":400,"altitudeEnd":null},{"regionName":"Vogelsbergkreis","start":1517907600000,"end":1517936400000,"level":2,"type":5,"state":"Hessen","description":"Es tritt leichter Frost zwischen 0 °C und -4 °C auf.","headline":"Amtliche WARNUNG vor FROST","event":"FROST","instruction":"","stateShort":"HE","altitudeStart":null,"altitudeEnd":null}],"103401000":[{"regionName":"Stadt Delmenhorst","start":1517914800000,"end":1517936400000,"level":2,"type":5,"state":"Niedersachsen","description":"Es tritt leichter Frost zwischen 0 °C und -4 °C auf.","headline":"Amtliche WARNUNG vor FROST","event":"FROST","instruction":"","stateShort":"NS","altitudeStart":null,"altitudeEnd":null}],"107132000":[{"regionName":"Kreis Altenkirchen","start":1517907600000,"end":1517936400000,"level":2,"type":5,"state":"Rheinland-Pfalz","description":"Es tritt oberhalb 400 m leichter Frost zwischen 0 °C und -4 °C auf.","headline":"Amtliche WARNUNG vor FROST","event":"FROST","instruction":"","stateShort":"RP","altitudeStart":400,"altitudeEnd":null},{"regionName":"Kreis Altenkirchen","start":1517907600000,"end":1517936400000,"level":2,"type":5,"state":"Rheinland-Pfalz","description":"Es tritt leichter Frost zwischen
...
```

I'm not sure, what's the correct way to fix this problem. A binding maybee could solve it more elegant than plain json and transforming it, but doesn't exists right now. 
Catching the exception in the rules seems to be another way to deal with it.

```
   try {
      // do some stuff
   }
   catch(Throwable t) {
      logError("Error", "Some bad stuff happened in my rule: " + T.toString)
   }
   finally {
      // always runs even if there was an error, good place for cleanup
   }
```

Any other suggestions welcome ...
Anyway .. The documentation should also be updated, because due to the change it's not correct for the actual 2.2. version.

Reply:
> I'm not sure, if #4870 addresses the same problem.

You are right, #4870 is addressing another problem - but it also had to return that "the value does not exist". So all I wanted to do is pointing you to an example how it can be done, i.e. return `UnDefType.NULL.toFullString()` as a valid state instead of `null`. 

> The documentation should also be updated

yes!